File: src/breslin/clientside/game/states/gamePlayPartido.cpp
//header
#include "gamePlayPartido.h"

//log
#include "../../tdreamsock/dreamSockLog.h"

//application
#include "../../application/applicationPartido.h"

//game
#include "../gamePartido.h"

//state machine
#include "../../../statemachine/stateMachine.h"

/***************************************
*	CONSTRUCTORS		          
***************************************/
GamePlayPartido::GamePlayPartido(GamePartido* gamePartido) : GamePlay(gamePartido)
{
	mGamePartido = gamePartido;
}

GamePlayPartido::~GamePlayPartido()
{

}

void GamePlayPartido::enter()
{
	LogString("GamePlayPartido::enter");
	GamePlay::enter();
}
void GamePlayPartido::execute()
{
	if (mGamePartido->mBattleStart)
	{
		mGamePartido->mStateMachine->changeState(mGamePartido->mGamePlayPartidoBattle);		
	}

        if (mGamePartido->mApplicationPartido->mGameReset)
        {
                mGamePartido->mStateMachine->changeState(mGamePartido->mGamePlayPartidoReset);
        }

	GamePlay::execute();
}

void GamePlayPartido::exit()
{
	GamePlay::exit();
}

File: src/breslin/clientside/game/states/game_play_partido.php
var GamePlayPartido = new Class(
{

Extends: GamePlay,


initialize: function(gamePartido)
{
	this.parent(gamePartido);
	this.mGamePartido = gamePartido; 
},

log: function(msg)
{
        setTimeout(function()
        {
                throw new Error(msg);
        }, 0);
},

enter: function()
{
	this.parent();
},

execute: function()
{
        if (this.mGamePartido.mBattleStart)
        {
                this.mGamePartido.mStateMachine.changeState(this.mGamePartido.mGamePlayPartidoBattle);
        }
        
        if (this.mGamePartido.mApplicationPartido.mGameReset)
        {
                this.mGamePartido.mStateMachine.changeState(this.mGamePartido.mGamePlayPartidoReset);
        }

	this.parent()
},	

exit: function()
{
	this.parent();
}

});

